src/main/java/org/folio/circulation/domain/validation/overriding/LoanValidator.java

```diff
@@ -0,0 +1,12 @@
+package org.folio.circulation.domain.validation.overriding;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.support.results.Result;
+
+public interface LoanValidator {
+
+  CompletableFuture<Result<LoanAndRelatedRecords>> validate(
```

Comment:
Let's change it to `validateAndHandleErrors` and also make these validators handle errors - the default implementation should handle the same error type that is handled now, but the override validator should handle `INSUFFICIENT_OVERRIDE_PERMISSIONS` type.